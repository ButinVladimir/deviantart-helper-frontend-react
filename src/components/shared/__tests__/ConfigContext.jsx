import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import consumeConfig, { getConfigProvider } from '../ConfigContext';
import defaultConfig from '../../../config';
import Config from '../../../config/config';

describe('ConfigContext', () => {
  const customComponent = ({ config, prop1, prop2 }) => (
    <div config={config}>
      {prop1}
      {prop2}
    </div>
  );
  customComponent.propTypes = {
    config: PropTypes.instanceOf(Config).isRequired,
    prop1: PropTypes.string.isRequired,
    prop2: PropTypes.string.isRequired,
  };

  it('can be rendered correctly', () => {
    const Provider = getConfigProvider();
    const props = {
      prop1: 'value',
      prop2: 'anotherValue',
    };

    const wrapper = mount(
      <Provider value={defaultConfig}>
        {consumeConfig(customComponent)(props)}
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
