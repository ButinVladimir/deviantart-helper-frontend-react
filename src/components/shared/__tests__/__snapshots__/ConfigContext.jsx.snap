// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigContext can be rendered correctly 1`] = `
<div
  config={
    Config {
      "serverUrl": "https://test.url/",
    }
  }
>
  value
  anotherValue
</div>
`;
